### DISCUSSION TITLE
Connecting ReactiveBase to local ElasticSearch DB

### DISCUSSION THREAD

**Discussion by kklhuang** [appbaseio/reactivesearch#2019] [Q&A] [answered]
Hi! I'm currently following the book search tutorial here: https://docs.appbase.io/docs/reactivesearch/v3/overview/reactivebase/ 
However, I'm trying to connect to my own local ElasticSearch DB on port 9200. 

I've referenced this issue discussion: https://github.com/appbaseio/reactivesearch/issues/104 

This is what I've changed: 

App.js: 
<img width="569" alt="Screen Shot 2022-07-07 at 1 39 43 PM" src="https://user-images.githubusercontent.com/56480921/177835910-3d2af227-19e8-4347-9ca8-16aee1f85d68.png">

elasticsearch.yml: 
<img width="1059" alt="Screen Shot 2022-07-07 at 1 40 11 PM" src="https://user-images.githubusercontent.com/56480921/177835980-7bbd6de8-59cf-4095-8706-3ac1cee9c088.png">

And this is my local db:
![Screen Shot 2022-07-07 at 1 41 02 PM](https://user-images.githubusercontent.com/56480921/177836141-e4a1e6ae-4838-4044-9080-7e85f0b079d6.png)

However, when I'm trying to get the data from each item to display on my localhost:3000 React page, nothing shows up. Is anyone able to help?

**Comment by siddharthlatest** [comment, marked as answer]
@kklhuang There are two options over here:

1. Set `enableAppbase` to `false` and your config should work as is. That's it!

OR

2. With `enableAppbase` (we should name this better), the query generation will happen on the server-side. It requires use of [reactivesearch-api](https://github.com/appbaseio/reactivesearch-api) binary, which acts as a gateway to Elasticsearch plus also handles server-side query generation. I highly recommend this for production usage so you're not exposing DSL on the frontend.

  So, you need to setup the reactivesearch-api binary. The easiest way to do this is via Docker. You can choose the [docker compose](https://github.com/appbaseio/reactivesearch-api-docker) file to use from here (comes in various combinations), or in the rare case where you're wanting to contribute to the project as well, here are the instructions to [build and run from source](https://github.com/appbaseio/reactivesearch-api#running-it).
  
  Once your binary is setup, then point the FE app's URL to it, e.g. `url=http://localhost:8000` instead of connecting to ES's port.
